### PR TITLE
Add NumberOf ctor with IText

### LINKED ISSUE #305: Add NumberOf ctor with IText
### Expected Behavior
Number has new ctor taking IText as argument

### PR DESCRIPTION
**Pull request by luplayer**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)
#305 

### What is the new behavior?
NumberOf has ctor for IText.
closes #305 

### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information

### CODE REVIEW

File: src/Yaapii.Atoms/Number/NumberOf.cs [hunk 1]
```diff
@@ -30,157 +30,230 @@ namespace Yaapii.Atoms.Number
     /// <summary>
     /// A parsed number
     /// </summary>
-    public sealed class NumberOf : NumberEnvelope
+    public sealed class NumberOf : INumber
```
Comment (koeeenig) [inline src/Yaapii.Atoms/Number/NumberOf.cs:33]: What`s the reason for this change?

Comment (luplayer) [reply src/Yaapii.Atoms/Number/NumberOf.cs:33]: There is no reason for this change. This change appeared due to the fact that i used the old version of this branch.

Comment (Meerownymous) [reply src/Yaapii.Atoms/Number/NumberOf.cs]: Please change back to using NumberEnvelope when possible.

**Review by Meerownymous — APPROVED** [review]

File: src/Yaapii.Atoms/Number/NumberOf.cs [hunk 2]
```diff
@@ -38,149 +38,173 @@ public sealed class NumberOf : NumberEnvelope
...
         { }
 
+
         /// <summary>
-        /// A <see cref="int"/> as a <see cref="INumber"/>
+        /// A <see cref="string"/> as a <see cref="INumber"/>
         /// </summary>
         /// <param name="str">The string</param>
         /// <param name="provider">a number format provider</param>
-        public NumberOf(string str, IFormatProvider provider) : this(str, new ScalarOf<IFormatProvider>(provider))
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Number/NumberOf.cs:97]: If you remove this ctor, it makes this a breaking change. So please preserve the ctor and add another one without a scalar if it is needed.

**Review by Meerownymous — CHANGES_REQUESTED** [review]

**Pull request merged**